internal/k8s/container.go

```diff
@@ -0,0 +1,96 @@
+package k8s
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/docker/distribution/reference"
+	"github.com/pkg/errors"
+	"github.com/windmilleng/tilt/internal/logger"
+	"k8s.io/api/core/v1"
+)
+
+func WaitForContainerReady(ctx context.Context, client Client, pod *v1.Pod, ref reference.Named) error {
+	ready, err := waitForContainerReadyHelper(pod, ref)
+	if err != nil {
+		return err
+	} else if ready {
+		return nil
+	}
+
+	watch, err := client.WatchPod(ctx, pod)
```

Comment:
no, the initial check is necessary for correctness. if we removed that check, and the container was already alive, then no changes would appear on the watch channel, and this would hang forever.

Reply:
added a test case for when the initial state of the pod is ok

Reply:
aaaah gotcha. nice!